test-suite/Math/NumberTheory/ModuliTests.hs

```diff
@@ -23,51 +23,51 @@ import Control.Arrow
 import Data.Bits
 import Data.List (tails, nub)
 import Data.Maybe
+import Data.Functor.Compose
 
 import Math.NumberTheory.Moduli
 import Math.NumberTheory.TestUtils
 
-toOdd :: Num a => a -> a
-toOdd n = n * 2 + 1
-
 unwrapPP :: (Prime, Power Int) -> (Integer, Int)
 unwrapPP (Prime p, Power e) = (p, e)
 
 -- | Check that 'jacobi' matches 'jacobi''.
-jacobiProperty1 :: (Integral a, Bits a) => AnySign a -> NonNegative a -> Bool
-jacobiProperty1 (AnySign a) (NonNegative (toOdd -> n)) = n == 1 && j == 1 || n > 1 && j == j'
+jacobiProperty1 :: (Integral a, Bits a) => AnySign a -> (Compose NonNegative Odd) a -> Bool
+jacobiProperty1 (AnySign a) (Compose (NonNegative (Odd n))) = n == 1 && j == 1 || n > 1 && j == j'
```

Comment:
Would view patterns or pattern synonyms help this code a teeny bit?
Either way cool use of compose aside from the orphanage issue :